lib/ash/actions/read/read.ex

```diff
@@ -236,7 +255,7 @@ defmodule Ash.Actions.Read do
           count,
           query.sort,
           query,
-          opts
+          Keyword.put(opts, :page, query.context[:page_opts] || opts[:page])
```

Comment:
I don't think we have to do this because we do this above `opts = Keyword.put(opts, :page, page_opts)`

Reply:
Makes sense